clang/test/Analysis/stream-error.c

```diff
@@ -299,6 +299,60 @@ void error_fseek_0(void) {
   fclose(F);
 }
 
+void error_fflush_0(void) {
+  FILE *F = tmpfile();
+  int Ret;
+  fflush(NULL);                      // no-warning
+  if (!F) {
+    if ((Ret = fflush(F)) != EOF)
+      clang_analyzer_eval(Ret == 0); // expected-warning {{TRUE}}
+    return;
+  }
+  if ((Ret = fflush(F)) != 0)
+    clang_analyzer_eval(Ret == EOF); // expected-warning {{TRUE}}
+  fclose(F);
+  fflush(F);                         // expected-warning {{Stream might be already closed}}
+}
+
+void error_fflush_1(void) {
```

Comment:
I prefer to have multiple tests (with a name that indicates what is tested) for the different cases.